echopype/clean/api.py

```diff
@@ -74,3 +85,206 @@ def remove_noise(ds_Sv, ping_num, range_sample_num, noise_max=None, SNR_threshol
+
+    mask = np.logical_not(mask)
+    return_mask = xr.DataArray(
+        mask,
+        dims=("ping_time", "range_sample"),
+        coords={"ping_time": source_Sv.ping_time, "range_sample": source_Sv.range_sample},
+    )
+    return return_mask
+
+
+def get_impulse_noise_mask(
+    source_Sv: xr.Dataset,
+    desired_channel: str,
+    thr: Union[Tuple[float, float], int, float],
+    m: Optional[Union[int, float]] = None,
+    n: Optional[Union[int, Tuple[int, int]]] = None,
+    erode: Optional[List[Tuple[int, int]]] = None,
+    dilate: Optional[List[Tuple[int, int]]] = None,
+    median: Optional[List[Tuple[int, int]]] = None,
+    method: str = "ryan",
+) -> xr.DataArray:
+    """
+    Algorithms for masking Impulse noise.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset
+        Dataset  containing the Sv data to create a mask
+    desired_channel: str
+        Name of the desired frequency channel.
+    thr: float or tuple
+        User-defined threshold value (dB) (ryan and ryan iterable) o
+        r a 2-element tuple specifying the range of threshold values (wang).
+    m: int or float, optional
+        Vertical binning length (in number of samples or range) (ryan and ryan iterable).
+        Defaults to None.
+    n: int or tuple, optional
+        Number of pings either side for comparisons (ryan),
+        or a 2-element tuple specifying the range (ryan iterable).
+        Defaults to None.
+    erode: List of 2-element tuples, optional
+        List indicating the window's size for each erosion cycle (wang). Defaults to None.
+    dilate: List of 2-element tuples, optional
+        List indicating the window's size for each dilation cycle (wang). Defaults to None.
+    median: List of 2-element tuples, optional
+        List indicating the window's size for each median filter cycle (wang). Defaults to None.
+    method: str, optional
+        The method (ryan, ryan iterable or wang) used to mask impulse noise. Defaults to 'ryan'.
+
+    Returns
+    -------
+    xr.DataArray
+        A DataArray consisting of a mask for the Sv data, wherein True values signify
+        samples that are free of noise.
+    """
+
+    # Our goal is to have a mask where True represents samples that are NOT impulse noise.
+    # So, we negate the obtained mask.
+
+    if method == "ryan":
+        impulse_mask_ryan = _find_impulse_mask_ryan(source_Sv, desired_channel, m, n, thr)
+        noise_free_mask = ~impulse_mask_ryan
+    elif method == "ryan_iterable":
+        impulse_mask_ryan_iterable = _find_impulse_mask_ryan_iterable(
+            source_Sv, desired_channel, m, n, thr
+        )
+        noise_free_mask = ~impulse_mask_ryan_iterable
+    elif method == "wang":
+        impulse_mask_wang = _find_impulse_mask_wang(
+            source_Sv, desired_channel, thr, erode, dilate, median
+        )
+        noise_free_mask = ~impulse_mask_wang
+    else:
+        raise ValueError(f"Unsupported method: {method}")
```

Comment:
I'd suggest the following:
- with the `parameters` dictionary added, add checks for if required parameters required by the specified `method` are provided
- create a mapping dictionary so that you can map the functions easily:
   ```
   MASK_MAP = {
      "ryan": _find_impulse_mask_ryan,
      "ryan_iterable": _find_impulse_mask_ryan_iterable,
      ...,
   }
   ```
   This way you'll be able to handle all functions in the same way and do the negation of mask in 1 line.

Reply:
https://github.com/OceanStreamIO/echopype/commit/759df07f772f7dcc624ea2565b7002267a4ed93c done